substrate/frame/people/src/lib.rs

```diff
@@ -0,0 +1,2059 @@
+					ring_index: base_ring_index,
+					ring_position: base_keys.len().saturated_into(),
+					scheduled_for_removal: false,
+				};
+				base_keys.try_push(key).map_err(|_| Error::<T>::TooManyMembers)?;
+				People::<T>::insert(personal_id, record)
+			}
+
+			// Newly added keys are not yet included.
+			RingKeys::<T>::insert(base_ring_index, base_keys);
+			RingKeysStatus::<T>::insert(base_ring_index, base_ring_status);
+			// Remove the stale ring root of the target ring. The keys in the target ring will be
+			// part of a valid ring root again when the base ring is rebuilt.
+			Root::<T>::remove(target_ring_index);
+			RingKeys::<T>::remove(target_ring_index);
+			RingKeysStatus::<T>::remove(target_ring_index);
+
+			Ok(())
+		}
+
+		// NOTE: This might be removed in the future. Use the `AsPerson` transaction
+		// extension to dispatch call with signature instead.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::as_personal_identity().saturating_add(call.get_dispatch_info().call_weight))]
+		pub fn as_personal_identity(
+			origin: OriginFor<T>,
+			index: PersonalId,
+			call: Box<<T as frame_system::Config>::RuntimeCall>,
+			signature: <T::Crypto as GenerateVerifiable>::Signature,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin.clone())?;
+			let record = People::<T>::get(index).ok_or(Error::<T>::NotPerson)?;
+			let key = record.key;
+			let ok = call.using_encoded(|msg| T::Crypto::verify_signature(&signature, msg, &key));
+			ensure!(ok, Error::<T>::InvalidSignature);
+			let derivation_weight = T::WeightInfo::as_personal_identity();
+			Self::derivative_call(origin, Origin::PersonalIdentity(index), *call, derivation_weight)
+		}
+
+		// NOTE: This might be removed in the future. Use the `AsPerson` transaction
+		// extension to dispatch call with signature instead.
+		/// Generally this should not be used since it does not protect against replay attacks. If
+		/// used then it is important to use mortal transactions with a short lifespan.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::as_personal_alias().saturating_add(call.get_dispatch_info().call_weight))]
+		pub fn as_personal_alias(
+			origin: OriginFor<T>,
+			context: Context,
+			call: Box<<T as frame_system::Config>::RuntimeCall>,
+			proof: <T::Crypto as GenerateVerifiable>::Proof,
+			ring_index: RingIndex,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin.clone())?;
+			let ring = Root::<T>::get(ring_index).ok_or(Error::<T>::NoMembers)?;
+			let alias = call
+				.using_encoded(|msg| T::Crypto::validate(&proof, &ring.root, &context, msg))
+				.map_err(|_| Error::<T>::InvalidProof)?;
+			let local_origin = Origin::PersonalAlias(RevisedContextualAlias {
+				revision: ring.revision,
+				ring: ring_index,
+				ca: ContextualAlias { alias, context },
+			});
+			let derivation_weight = T::WeightInfo::as_personal_alias();
+			Self::derivative_call(origin, local_origin, *call, derivation_weight)
+		}
+
```

Comment:
Maybe better to remove them and introduce something specific when we see the need.

Because for now, they don't protect against replay. The proof can be reused forever.
There is no `with_mortality` call in utility that can be used to call into to have some replay protection.

```suggestion
```

Reply:
I agree.